crates/configuration/src/version4/mod.rs

```diff
@@ -405,549 +419,93 @@ fn base_type_representations() -> BTreeMap<String, database::TypeRepresentation>
-            .map(|(k, column_info)| (k, convert_column_info(column_info)))
-            .collect(),
-        uniqueness_constraints: convert_uniqueness_constraints(table_info.uniqueness_constraints),
-        foreign_relations: convert_foreign_relations(table_info.foreign_relations),
-        description: table_info.description,
-    }
-}
-
-fn convert_foreign_relations(
-    foreign_relations: metadata::ForeignRelations,
-) -> query_engine_metadata::metadata::ForeignRelations {
-    query_engine_metadata::metadata::ForeignRelations(
-        foreign_relations
-            .0
-            .into_iter()
-            .map(|(k, foreign_relation)| (k, convert_foreign_relation(foreign_relation)))
-            .collect(),
+    // create the configuration file
+    fs::write(
+        configuration_file,
+        serde_json::to_string_pretty(&parsed_config)
+            .map_err(|e| WriteParsedConfigurationError::IoError(e.into()))?
+            + "\n",
     )
-}
+    .await?;
 
-fn convert_foreign_relation(
-    foreign_relation: metadata::ForeignRelation,
-) -> query_engine_metadata::metadata::ForeignRelation {
-    query_engine_metadata::metadata::ForeignRelation {
-        foreign_schema: foreign_relation.foreign_schema,
-        foreign_table: foreign_relation.foreign_table,
-        column_mapping: foreign_relation.column_mapping,
+    // look for native query sql file references and write them to disk.
+    for native_query_sql in parsed_config.metadata.native_queries.0.values() {
+        if let metadata::NativeQuerySqlEither::NativeQuerySql(
+            metadata::NativeQuerySql::FromFile { file, sql },
+        ) = &native_query_sql.sql
+        {
+            if file.is_absolute() || file.starts_with("..") {
+                Err(
+                    WriteParsedConfigurationError::WritingOutsideDestinationDir {
+                        dir: out_dir.as_ref().to_owned(),
+                        file: file.clone(),
+                    },
+                )?;
+            };
+
+            let native_query_file = out_dir.as_ref().to_owned().join(file);
+            if let Some(native_query_sql_dir) = native_query_file.parent() {
+                fs::create_dir_all(native_query_sql_dir).await?;
+            };
+            fs::write(native_query_file, String::from(sql.clone())).await?;
+        };
     }
-}
 
-fn convert_uniqueness_constraints(
-    uniqueness_constraints: metadata::UniquenessConstraints,
-) -> query_engine_metadata::metadata::UniquenessConstraints {
-    query_engine_metadata::metadata::UniquenessConstraints(
-        uniqueness_constraints
-            .0
-            .into_iter()
-            .map(|(k, uniqueness_constraint)| {
-                (k, convert_uniqueness_constraint(uniqueness_constraint))
-            })
-            .collect(),
+    // create the jsonschema file
+    let configuration_jsonschema_file_path = out_dir
+        .as_ref()
+        .to_owned()
+        .join(CONFIGURATION_JSONSCHEMA_FILENAME);
```

Comment:
Wondering if each version should have a unique schema.json filename, like `schema-v3.json`, `schema-v4.json`, etc...

Reply:
As long as `configuration.json` indicates which schema file to use then I suppose we could this.

I wouldn't think too hard about this until we take on actually intentionally designing the configuration format.